Refactor: Use externalAttributionSources instead of signals to label SPDX elements

meretp opened the pull request:
Similar to the opossum file generated by the Haskell tooling, this PR adds `externalAttributionSources` to label the SPDX elements (file, package or snippet). Previously we added an additional signal for this (as shown in the screenshot below). 
![image](https://github.com/opossum-tool/opossum.lib.py/assets/50019307/ba5fe042-1c95-4119-95d2-49cb38eea53a)

With these changes the SPDXID of each element is no longer visible but instead all packages / files / snippets included in an SPDX element are grouped under "SPDX-Package" / "SPDX-File" / "SPDX_Snippet" in the list of signals (see the screenshot below).  
![image](https://github.com/opossum-tool/opossum.lib.py/assets/50019307/3f398d6b-1543-400e-bf80-9c2dd11d9d64)

JakobSchubert commented on src/opossum_lib/attribution_generation.py:
```diff
@@ -10,7 +10,7 @@
 
 
 def create_package_attribution(package: Package) -> OpossumPackage:
-    source = SourceInfo(package.spdx_id)
```
Might be good to extract a constant for "SPDX-Package" and also use that in file_generation.py . Otherwise somebody changing this string in the future might forget to also adapt file_generation.py
.
Same applies to SPDX-File and SPDX-Snippet.

JakobSchubert reviewed (approved):
Nice, I really like this change. The Opossum file now looks even better. Also thanks for the before-after Screenshots in the description.

Just one small comment.

meretp commented:
In #44 I added a `constants` module, to be able to use this module and add the constants extracted in this PR, I rebased and force-pushed this branch.

JakobSchubert commented:
:+1: great, feel free to merge.
